### DISCUSSION TITLE
v2.2.0 (Beta)

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#2716] [Releases]
## Installation
Visit [https://getutm.app/install/][1] for the most up to date installation instructions.

## New Features
* (macOS 11.3+ Only) **Bridged networking and shared networking support**. You can configure it in the Network settings for your VM.
* (macOS 11+ and iOS 13+) **GPU acceleration for OpenGL on Linux**. Use `virtio-ramfb` or `virtio-gpu-gl-pci` display device and compatible Linux drivers (most modern Linux distros will have it already installed). Windows is not supported because there is currently no virtio-gpu driver for Windows that supports 3D acceleration.
* QEMU is updated to v6.1.0-rc1

## Known Issues
* On Intel Macs, `virtio-ramfb` and `virtio-gpu-pl-pci` does not work with BIOS boot. You need to boot from UEFI which currently requires you to manually add a PC Flash device with the UEFI image. In the future, there will be an option to boot from UEFI instead of BIOS.
* On macOS 12, some UI elements are laid out incorrectly due to changes in SwiftUI apis.

Please check the full list on [Github][2] and help report any bug you find that is not listed.

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v2.2.0'>v2.2.0 (Beta)</a>.</em>

**Comment by ktprograms** [comment]
**Describe the issue**
Getting an error when starting a new VM: `qemu-aarch64-softmmu: failed to initialize spice server`.
This error shows up no matter which network mode is used (including `None`).

**Configuration**
* UTM Version: v2.2.0 (Beta)
* OS Version: macOS Big Sur 11.5.1
* Intel or Apple Silicon? Apple Silicon

**Crash log**
None

**Debug log**
[debug.log](https://github.com/utmapp/UTM/files/6952377/debug.log)

**Upload VM**
[config.plist.zip](https://github.com/utmapp/UTM/files/6952386/config.plist.zip)

**Reply by osy** [reply]
Ah I see the error, it’s in the build.

**Reply by ktprograms** [reply]
Just out of interest, what's the error?

**Reply by osy** [reply]
The build script running on GitHub isn’t registered with my teamid so it’s generating the wrong container URL.

**Reply by VishuPillu** [reply]
How to enable camera in emulator?

**Comment by ktprograms** [comment]
BTW setting custom icons works in this version. #2387

**Comment by conath** [comment]
My existing Windows 10 x86_64 VM doesn't work anymore with this version, even though I have not changed anything and had already set the system to remain the 6.0 one in the last version. The problem is as follows:
the first boot worked fine, then it crashed after started up completely and restarted. Then it failed to get past Windows boot so went into `Preparing Automatic Repair`. That also fails because the hard drive is `viostor`, the driver for which is missing in repair mode.

Maybe, did something with the hard disks change?

When trying to reinstall from ISO it keeps crashing before the Windows loading spinner appears. 
[debug.log](https://github.com/utmapp/UTM/files/6956652/debug.log)

Update: when I change back to 2.1.2 it boots fine again. Log from 2.1.2: 
[debug-2.1.2.log](https://github.com/utmapp/UTM/files/6956687/debug-2.1.2.log)

**Comment by conath** [comment]
I've been playing with the new openGL accelerated `virtio-ramfb` in Ubuntu Server 20.04 ARM. It's really nice!

YouTube playback at 1080p without any lag and smooth UI at Retina resolutions. 😍

**Reply by ktprograms** [reply]
Is there any specific settings needed? `glxinfo -B` prints this:
```
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Mesa/X.org (0x1af4)
    Device: virgl (0x1010)
    Version: 20.2.6
    Accelerated: yes
    Video memory: 0MB
    Unified memory: no
    Preferred profile: compat (0x2)
    Max core profile version: 0.0
    Max compat profile version: 2.1
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.0
OpenGL vendor string: Mesa/X.org
OpenGL renderer string: virgl
OpenGL version string: 2.1 Mesa 20.2.6
OpenGL shading language version string: 1.20

OpenGL ES profile version string: OpenGL ES 3.0 Mesa 20.2.6
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.00
```
It seems to only have the `compatibility` and `ES` profiles. Also, `glmark2` fails (Seg fault, maybe [this?](https://github.com/glmark2/glmark2/issues/124)) on the fifth test and `glmark2-es2` randomly locks up the screen (cpu is running and can `ssh` in, but only the cursor moves).

**Reply by conath** [reply]
I didn’t change any settings in the guest.

**Reply by ktprograms** [reply]
What's your `glxinfo`?

**Reply by conath** [reply]
```bash
conath@conaths-ubuntu-server:~$ glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Mesa/X.org (0x1af4)
    Device: virgl (0x1010)
    Version: 21.0.3
    Accelerated: yes
    Video memory: 0MB
    Unified memory: no
    Preferred profile: compat (0x2)
    Max core profile version: 0.0
    Max compat profile version: 2.1
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.0
OpenGL vendor string: Mesa/X.org
OpenGL renderer string: virgl
OpenGL version string: 2.1 Mesa 21.0.3
OpenGL shading language version string: 1.20

OpenGL ES profile version string: OpenGL ES 3.0 Mesa 21.0.3
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.00
```

**Reply by ktprograms** [reply]
Can you run `glmark2` successfully?

**Reply by conath** [reply]
No, I also get a segfault:
![GLMark2](https://user-images.githubusercontent.com/12073163/129008594-13ac927a-8a21-4eda-9776-de00c0bf8e25.jpg)

**Reply by ktprograms** [reply]
What about `glmark2-es2`?

**Reply by conath** [reply]
It ran well for a while and then the entire VM hung, QEMULauncher process at 50 GB virtual memory. 
![GLMark2es2-hang](https://user-images.githubusercontent.com/12073163/129011089-d16f5e80-df23-471c-a4fa-e537a10bca2e.jpg)

**Reply by ktprograms** [reply]
I didn't check the virtual memory, but pretty much the same happened to me. Do you think an issue should be opened?

**Reply by conath** [reply]
There is already #2720 about the memory usage. If you want you can open a separate one about the compatibility.

**Comment by tomkoi** [comment]
@osy how can i download the beta version for the MacOS?
i really want to try the bridge network option

**Reply by osy** [reply]
Check the releases.

**Reply by tomkoi** [reply]
> Check the releases.

not sure i got it.
i tried to find some place here to get the link for the download and didn't found any place.
can you share please the link?

**Reply by ktprograms** [reply]
https://github.com/utmapp/UTM/releases

**Reply by ktprograms** [reply]
You'll probably want v2.2.1 (Beta)

**Reply by tomkoi** [reply]
@ktprograms 
thanks but i will be happy to try version 2.2.0

**Reply by ktprograms** [reply]
v2.2.0 has a bug that the emulated graphics uses too much memory. It was fixed in v2.2.1 and the bridged network is still available in v2.2.1

**Reply by tomkoi** [reply]
> v2.2.0 has a bug that the emulated graphics uses too much memory. It was fixed in v2.2.1 and the bridged network is still available in v2.2.1

ok, and how i can download it?

**Reply by conath** [reply]
Here is a direct link: [UTM.dmg (2.2.1)](https://github.com/utmapp/UTM/releases/download/v2.2.1/UTM.dmg)

**Reply by tomkoi** [reply]
> Here is a direct link: [UTM.dmg (2.2.1)](https://github.com/utmapp/UTM/releases/download/v2.2.1/UTM.dmg)

thanks!!!! npt sure why i didn't see
